In [ ]:
######################## 29-08-2024 ##############################

In [ ]:
# import the packages 
#read the data
#convert categorical and numerical columns using select datatypes

In [2]:
import pandas as pd
visa_df=pd.read_csv('Visadataset.csv')
visa_df

case_id continent education_of_employee has_job_experience  \
0         EZYV01      Asia           High School                  N   
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
4         EZYV05    Africa              Master's                  Y   
...          ...       ...                   ...                ...   
25475  EZYV25476      Asia            Bachelor's                  Y   
25476  EZYV25477      Asia           High School                  Y   
25477  EZYV25478      Asia              Master's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
1                         N             2412         2002   
2                         Y            44444         2008   
3                         N               98         1897   
4                         N             1082         2005   
...                     ...              ...          ...   
25475                     Y             2601         2008   
25476                     N             3274         2006   
25477                     N             1121         1910   
25478                     Y             1918         1887   
25479                     N             3195         1960   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
1                Northeast       83425.6500         Year                  Y   
2                     West      122996.8600         Year                  Y   
3                     West       83434.0300         Year                  Y   
4                    South      149907.3900         Year                  Y   
...                    ...              ...          ...                ...   
25475                South       77092.5700         Year                  Y   
25476            Northeast      279174.7900         Year                  Y   
25477                South      146298.8500         Year                  N   
25478                 West       86154.7700         Year                  Y   
25479              Midwest       70876.9100         Year                  Y   

      case_status  
0          Denied  
1       Certified  
2          Denied  
3          Denied  
4       Certified  
...           ...  
25475   Certified  
25476   Certified  
25477   Certified  
25478   Certified  
25479   Certified  

[25480 rows x 12 columns]

In [3]:
cat_cols=visa_df.select_dtypes(include='object').columns        #  the names of the column who contains charectoes or alphabets
num_cols=visa_df.select_dtypes(exclude='object').columns        #  the names of the column who contains numerical values
# by using this we can take numerical columns and categorical columns

In [4]:
cat_cols

Index(['case_id', 'continent', 'education_of_employee', 'has_job_experience',
       'requires_job_training', 'region_of_employment', 'unit_of_wage',
       'full_time_position', 'case_status'],
      dtype='object')

In [4]:
num_cols

Index(['no_of_employees', 'yr_of_estab', 'prevailing_wage'], dtype='object')

**how many ways retrive a column**

- i want to retrive a continent column

In [5]:
visa_df['continent'] #series              # print the series of 'continent'
visa_df[['continent']]  #DataFrame
visa_df.iloc[:,1]         #series         # it gives the series of perticular column and value ,it is for integer only 
visa_df.loc[:,'continent'] #series
visa_df.iloc[:,[1]] # Dataframe
visa_df.loc[:,['continent']] #Dataframe
visa_df.continent #series


0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
          ...  
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia
Name: continent, Length: 25480, dtype: object

**unique**

- read the continent column apply unique methood

- it will give the continents lables which are repeating entire data

In [6]:
visa_df['continent'].unique()

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

- total dataro are 25480
- continent has only 6 unique labels

**nununique**

In [7]:
visa_df['continent'].nunique()

6

**task:1**

- we know that only 6 labels are reapiting in entire 25480 observations

- now we want to know how many times **Asia** reapiting

- we already know that cat column can represent using **frequency table**

- our main goal is to find the frequency table

In [21]:
# getting the count of total no of application from each continent

# step-:read continent data
# step-2 we want asia so the dataequal to asia 

visa_df['continent']                     # first we took the continent column
con=visa_df['continent']=='Asia'        # in continent we took asia
# print(con)
visa_df[con]                     # then we took the data from asia 
len(visa_df[con])               # after that we count the length of asia 



visa_df['continent']
con=visa_df['continent']=='Africa'
visa_df[con]
print(len(visa_df[con]))

visa_df['continent']
con=visa_df['continent']=='North America'
visa_df[con]
print(len(visa_df[con]))

visa_df['continent']
con=visa_df['continent']=='Europe'
visa_df[con]
print(len(visa_df[con]))

visa_df['continent']
con=visa_df['continent']=='South America'
visa_df[con]
print(len(visa_df[con]))

visa_df['continent']
con=visa_df['continent']=='Oceania'
visa_df[con]
print(len(visa_df[con]))


# found h

16861

In [29]:
# by using the for loop getting the count of total no of application from each continent

# printing the count
cont_lables=visa_df['continent'].unique()
for i in cont_lables:
    visa_df['continent']
    con=visa_df['continent']==i
    visa_df[con]
    count=len(visa_df[con])
    print(f'The number of applicants from {i} is : {count}')

# counted the number of applications in verious continents

print("this is instruction")


The number of applicants from Asia is : 16861
The number of applicants from Africa is : 551
The number of applicants from North America is : 3292
The number of applicants from Europe is : 3732
The number of applicants from South America is : 852
The number of applicants from Oceania is : 192
this is instruction


array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

In [37]:
# appending the count in list
cont_labels=visa_df['continent'].unique()
count=[]
for i in count_labels:
    visa_df['continent']
    visa_df['continent']==i
    visa_df[con]
    count.append( len(visa_df[con]))
count,cont_lables

#created a list of no.of applicant and continents

([192, 192, 192, 192, 192, 192],
 array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
        'Oceania'], dtype=object))

In [38]:
pd.DataFrame(zip(cont_lables,count),
            columns=['Continent','Number of Applicants'])    # after creating the list, created a dataframe Dataframe 

Continent  Number of Applicants
0           Asia                   192
1         Africa                   192
2  North America                   192
3         Europe                   192
4  South America                   192
5        Oceania                   192

In [24]:
visa_df['continent'].values_column()

AttributeError: 'Series' object has no attribute 'values_column'

In [ ]:
# paralell bank data set 

In [ ]:
########################################### 30-08-2024 #####################################

In [27]:
# import the packages
# read the data
# convert into cat and num colmsd

import pandas as pd

visa_df=pd.read_csv('Visadataset.csv')
# cat_cols=visa_df.select_dtypes(include='object').columns
num_cols=visa_df.select_dtypes(exclude='object').columns
num_cols

Index(['no_of_employees', 'yr_of_estab', 'prevailing_wage'], dtype='object')

In [ ]:
# we learned how to read the 

In [3]:
count_data=visa_df['continent'].value_counts()
#series
# when we convert series into dictionary key value pair

continent
Asia             16861
Europe            3732
North America     3292
South America      852
Africa             551
Oceania            192
Name: count, dtype: int64

In [7]:
con_keys=visa_df['continent'].value_counts().keys()

In [6]:
count_values=visa_df['continent'].value_counts().values

In [9]:
cols=['continent','Number of Applications']
pd.DataFrame(zip(con_keys,count_values),columns=cols)

continent  Number of Applications
0           Asia                   16861
1         Europe                    3732
2  North America                    3292
3  South America                     852
4         Africa                     551
5        Oceania                     192

In [10]:
count_values=con_keys.values()
count_values                   # by second method

TypeError: 'numpy.ndarray' object is not callable

In [ ]:
method 2 using the for loop

In [ ]:
count_labels=visa_df['continent'].unique()
count=[]
for i in count_labels:
    visa_df['continent']==i
    visa_df[con]
    count.append

In [ ]:
count_df1.to_csv('continent_info.csv',index=False)
count_df2.to_excel('continet_info.xlsx',index=False)

**task 2**

- Create all categorical columns frequency table using value_counts

- using python create one directory (folder) with the name **frequency table**

- you need to iterate each column, but each has different labels

- provide one column name as **label**

- anther colimn as **count**

In [19]:
cont_data=visa_df['continent'].value_counts()
con_keys=cont_data.keys()
cont_values=cont_data.values
cols=['continents','Number of applicants']
cont_df1=pd.DataFrame(zip(con_keys,cont_values),columns=cols)
cont_df1.to_csv('continent_info.csv',index=False)

In [23]:
for i in cat_cols[1:]:
    print(f'{i}_info.csv')

yr_of_estab_info.csv
prevailing_wage_info.csv


**create a folder**

- Step 1: get the current working directory

- step 2: what is your new folder name

- step 3: join both

- Step 4: make the directory

In [31]:
import os 
cwd = os.getcwd()
folder_name = 'frequency_table_by_me'
new_dir=os.path.join(cwd,folder_name)
try:
    os.makedirs(new_dir)
except Exception as e:
    print(e)

[WinError 183] Cannot create a file when that file already exists: 'C:\\Users\\DELL\\Data Science Python\\frequency_table_by_me'


In [33]:
for i in cat_cols[1:]:
    print(f'{new_dir}\\{i}_info.csv')

C:\Users\DELL\Data Science Python\frequency_table_by_me\yr_of_estab_info.csv
C:\Users\DELL\Data Science Python\frequency_table_by_me\prevailing_wage_info.csv


In [32]:
for i in cat_cols[1:]:
    data=visa_df[i].values_count()
    keys=data.keys()
    

In [ ]:
################## waiting for session 3 update #####################
^
|

############################################################### 01-09-2024 #############################################################

In [24]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
# import 

visa_df=pd.read_csv('Visadataset.csv')
cat_cols=visa_df.select_dtypes(include='object').columns
num_cols=visa_df.select_dtypes(include='object').columns

In [21]:
# frequency table 
# we have two methods
# method1 : using the unique and for loop 
# method 2: using the value count (method)
import pandas as pd
visa_df=pd.read_csv('Visadataset.csv')
visa_df
cont_data = visa_df['continent'].value_counts()
con_keys = cont_data.keys()
cont_values = cont_data.values
cols=['continent','number of applications']
cont_df1=pd.DataFrame(zip(cont_data,con_keys,cont_values),columns=cols)
cont_df1

# the above task we need to repeate for all cat columns
# but in that we are avoiding case id: many unique values
# the above line 

continent number of applications      A
0      16861                   Asia  16861
1       3732                 Europe   3732
2       3292          North America   3292
3        852          South America    852
4        551                 Africa    551
5        192                Oceania    192

In [17]:
import os 
folder='frequency table_2'
new_dir=os.path.join(os.getcwd(),folder)
os.makedirs(new_dir)